pommerman/graphics.py

```diff
@@ -0,0 +1,333 @@
+
+
+    def set_agents(self, agents):
+        self._agents = agents
+
+
+    def render(self):
+        viewer = self._viewer
+
+
+    def save(self, path):
+        from PIL import Image
+        img = []
+        Image.fromarray(img).save(path)
+
+
+class PommeViewer(object):
+    def __init__(self, 
+                display=None, 
+                board_size=13, 
+                agents=[], 
+                partially_observable=False,
+                game_type=None
+    ):
+        display = rendering.get_display(display)
+        board_height = TILE_SIZE * board_size
+        height =  math.ceil(board_height + (BORDER_SIZE * 2) + (MARGIN_SIZE * 3))
+        width = math.ceil(board_height + board_height / 4 + (BORDER_SIZE * 2) + MARGIN_SIZE)
+
+        self._height = height
+        self._width = width
+        self.window = pyglet.window.Window(width=width, height=height, display=display)
+        self.window.set_caption('Pommerman')
+        self.isopen = True
+        self._board_size = board_size
+        self._board_state = None
+        self._resource_manager = ResourceManager()
+        self._tile_size = TILE_SIZE
+        self._agent_tile_size =  (board_height / 4) / board_size
+        self._agent_count = len(agents)
+        self._agents = agents
+        self._step = 0
+        self._game_type = game_type
+        self._is_partially_observable = partially_observable
+        self._agent_view_size = 4
+
+        glEnable(GL_BLEND)
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA)
+
+        @self.window.event
+        def close(self):
+            self.window.close()
+            self.isopen = False
+
+    def set_agent_view_size(self, view_size):
+        self._agent_view_size = view_size
+
+    def set_agents(self, agents):
+        self._agents = agents
+    
+    def set_board(self, state):
+        self._board_state = state
+
+    def set_step(self, step):
+        self._step = step
+
+
+    def render(self):
+        self.window.switch_to()
+        self.window.dispatch_events()
+        
+        self.render_background()
+        self.render_text()
+        self.render_dead_alive()
+
+        self.render_main_board()
+        self.render_agents_board()
+        
+        self.window.flip()
+
+
+    def render_main_board(self):
+        board = self._board_state
+        size = self._tile_size
+        x_offset = BORDER_SIZE
+        y_offset = BORDER_SIZE
+        self.render_board(board, x_offset, y_offset, size)
+
+
+    def render_agents_board(self):
+        x_offset = self._board_size * self._tile_size + BORDER_SIZE + MARGIN_SIZE
+        size = self._agent_tile_size
+        for agent in self._agents:
+            y_offset = agent.agent_id * size * self._board_size + ( agent.agent_id * MARGIN_SIZE ) + BORDER_SIZE
+            agent_board = self.agent_view(agent)
+            self.render_board(agent_board, x_offset, y_offset, size)
+
+
+    def render_board(self, board, x_offset, y_offset, size):
```

Comment:
Should we be blitting each one or batching (http://pyglet.readthedocs.io/en/pyglet-1.3-maintenance/modules/graphics/index.html#pyglet.graphics.Batch) them together a la https://stackoverflow.com/questions/46044714/pyglet-loading-blitting-image-with-alpha?